src/mlir_build.cpp

```diff
@@ -76,25 +89,25 @@ static void MlirGenerateSourcesForHipBuild(const boost::optional<TmpDir>& tmp_di
 
     MIOPEN_LOG_I2(input_file.string() << ", options: '" << params << "'");
 
-    AutoMlirHandle handle(params);
-    MlirLowerCpp(handle());
+    AutoMiirHandle handle(params);
+    miirLowerCpp(handle());
```

Comment:
`miirLowerInit()` is missing here.

Reply:
`miirLowerCpp()` is the only lowering pipeline that does not need any llvm passes. We hacked out the pass such that it populate cpp, rather than other sane products such as llvm ir. The more I speak of it only make me to hate it more. Sorry about the confusions for it caused. We should just leave the cpp backend as-is for now.

Reply:
NP! This is experimental.